docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+      - [4.3.1. Addition of chore](#431-addition-of-chore)
+      - [4.3.2. List all/ specific chore(s)](#432-list-all-specific-chores)
+      - [4.3.3. Delete all/ specific chore(s)](#433-delete-all-specific-chores)
+      - [4.3.4. Search for chore based on keyword(s)](#434-search-for-chore-based-on-keywords)
+    + [4.4. Storage](#44-storage)
+      - [4.4.1. Select files to load from and save to](#441-select-files-to-load-from-and-save-to)
+      - [4.4.2. Save current state](#442-save-current-state)
+    + [4.5. Logging](#45-logging)
+    + [4.6. Configuration](#46-configuration)
+  * [Appendices](#appendices)
+    + [Appendix A: Product Scope](#appendix-a-product-scope)
+    + [Appendix B: User Stories](#appendix-b-user-stories)
+    + [Appendix C: Value proposition - Use cases](#appendix-c-value-proposition---use-cases)
+    + [Appendix D: Non-Functional Requirements](#appendix-d-non-functional-requirements)
+    + [Appendix E: Glossary](#appendix-e-glossary)
+    + [Appendix F: Instructions for Manual Testing](#appendix-f-instructions-for-manual-testing)
+      - [F.1. Launch and Shutdown](#f1-launch-and-shutdown)
+      - [F.2. Add an ingredient](#f2-add-an-ingredient)
+      - [F.3. Search for ingredient](#f3-search-for-ingredient)
+      - [F.4. Search for recipe](#f4-search-for-recipe)
+      - [F.5. Search for chore](#f5-search-for-chore)
+      - [F.6. Saving data](#f6-saving-data)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction
+### 1.1. Purpose
+The document contains the specified architecture and software design specifications for the application, Kitchen Helper. 
+### 1.2. Scope
+This describes the software architecture and software design requirements for Kitchen Helper. This guide is mainly for developers, designers and software engineers that are or going to work on Kitchen Helper. 
+## 2. Setting up
 
+### 2.1. Prerequisites
+1. JDK `11`.
+2. IntelliJ IDE.
 
-## Product Scope
-### Target user profile
+### 2.2. Setting up the project in your computer
+1. Fork this repository, and clone the fork repository to your computer.
+2. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   1. Click `Configure` > `Structure for New Projects` and then `Project Settings` > `Project` > `Project SDK`.
+   2. If `JDK 11` is listed in the drop down, select it. Otherwise, click `New…` and select the directory where you installed `JDK 11`.
+   3. Click `OK`.
+4. Click `Import Project`.
+5. Locate the `build.gradle` file and select it. Click `OK`.
+6. Click `Open as Project`.
+7. Click `OK` to accept the default settings if prompted. 
 
-{Describe the target user profile}
+## 3. Design
+This section provides a high level overview of our application, Kitchen Helper.
+### 3.1. Architecture
```

Comment:
Like the other two comments, I agree that the architecture is well designed! 👍  :)